hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1725,6 +1729,173 @@ public void delete(final Path path, final boolean recursive,
     } while (shouldContinue);
   }
 
+  private void deleteBlobPath(final Path path,
+      final boolean recursive,
+      final TracingContext tracingContext) throws IOException {
+    StringBuilder listSrcBuilder = new StringBuilder();
+    listSrcBuilder.append(path.toUri().getPath());
+    if (!path.isRoot()) {
+      listSrcBuilder.append(FORWARD_SLASH);
+    }
+
+    String listSrc = listSrcBuilder.toString();
+    BlobList blobList = client.getListBlobs(null, listSrc, null, null,
+        tracingContext).getResult().getBlobList();
+    if (blobList.getBlobPropertyList().size() > 0) {
+      LOG.debug(
+          String.format("Path %s is child-blobs", path.toUri().getPath()));
```

Comment:
The statement looks gramatically incorrect.

Reply:
Done